crates/core/src/operations/writer.rs

```diff
@@ -218,18 +218,11 @@ impl DeltaWriter {
     /// This will flush all remaining data.
     pub async fn close(mut self) -> DeltaResult<Vec<Add>> {
         let writers = std::mem::take(&mut self.partition_writers);
-        let actions = futures::stream::iter(writers)
-            .map(|(_, writer)| async move {
-                let writer_actions = writer.close().await?;
-                Ok::<_, DeltaTableError>(writer_actions)
-            })
-            .buffered(num_cpus::get())
-            .try_fold(Vec::new(), |mut acc, actions| {
-                acc.extend(actions);
-                futures::future::ready(Ok(acc))
-            })
-            .await?;
-
+        let mut actions = Vec::new();
+        for (_, writer) in writers {
+            let writer_actions = writer.close().await?;
+            actions.extend(writer_actions);
+        }
```

Comment:
@Dan-J-D I'm not sure I understand the motivation of this change. I believe @alexwilcoxson-rel or @wjones127 introduced the original code above to ensure that the process which was running the writer would be able to concurrently produce the `Vec<Action>`

Based on my understanding of this change, it walks back that performance improvement but I cannot figure out for what gain :thinking:

Reply:
I was on a older version of the library and I just copy & pasted my local copy into github, but there were newer changes already made, so I added them back